lens/util/repo.go

```diff
@@ -340,3 +344,125 @@ func MakeGetActorCodeFunc(ctx context.Context, store adt.Store, next, current *t
 		return act.Code, true
 	}, nil
 }
+
+type marshaller func(interface{}) ([]byte, error)
+
+func MarshalWithOverrides(v interface{}, overrides map[reflect.Type]marshaller) ([]byte, error) {
+	pwt := paramWrapperType{
+		obj:     v,
+		replace: overrides,
+	}
+	return pwt.MarshalJSON()
+}
+
+// wrapper type for overloading json marshal methods
+type paramWrapperType struct {
+	obj     interface{}
+	replace map[reflect.Type]marshaller
+}
+
+func (wt *paramWrapperType) MarshalJSON() ([]byte, error) {
+	v := reflect.ValueOf(wt.obj)
+	t := v.Type()
+
+	// if this is the type we want to override marshalling for, do the thing.
+	rf, ok := wt.replace[t]
+	if ok {
+		return rf(wt.obj)
+	}
+
+	// if the type has its own marshaller use that
+	if t.Implements(reflect.TypeOf((*json.Marshaler)(nil)).Elem()) {
+		return json.Marshal(wt.obj)
+	}
+
+	switch t.Kind() {
+	case reflect.Ptr:
+		// unwrap pointer
+		v = v.Elem()
+		t = t.Elem()
+		fallthrough
+
+	case reflect.Struct:
+		// if its a struct, walk its fields and recurse.
+		m := make(map[string]interface{})
+		for i := 0; i < v.NumField(); i++ {
+			if t.Field(i).IsExported() {
+				m[t.Field(i).Name] = &paramWrapperType{
+					obj:     v.Field(i).Interface(),
+					replace: wt.replace,
+				}
+			}
+		}
+		return json.Marshal(m)
+
+	case reflect.Slice:
+		// if its a slice of go types, marshal them, otherwise walk its indexes and recurse
+		var out []interface{}
+		if v.Len() > 0 {
+			switch v.Index(0).Kind() {
+			case
+				reflect.Bool,
+				reflect.String,
+				reflect.Map,
+				reflect.Float32, reflect.Float64,
+				reflect.Complex64, reflect.Complex128,
+				reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64,
+				reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+				return json.Marshal(v.Interface())
+			default:
+			}
+		}
+		for i := 0; i < v.Len(); i++ {
+			out = append(out, &paramWrapperType{
+				obj:     v.Index(i).Interface(),
+				replace: wt.replace,
+			})
+		}
+		return json.Marshal(out)
+
+	default:
+		return json.Marshal(wt.obj)
+	}
+}
+
+// marshal go-bitfield to json with count value included.
+var bitfieldCountMarshaller = func(v interface{}) ([]byte, error) {
+	rle := v.(bitfield.BitField)
+	r, err := rle.RunIterator()
+	if err != nil {
+		return nil, err
+	}
+	count, err := rle.Count()
+	if err != nil {
+		return nil, err
+	}
+
+	var ret = struct {
+		Count uint64
+		RLE   []uint64
+	}{}
```

Comment:
Given:

> [Old](https://github.com/filecoin-project/lily/issues/819#issue-1113354802):
`{"Recoveries": [{"Sectors": {"rle": "[10816,31,1407,1,2,23,1,4,1,1,1405,1,3,26,1,1,1,1,521,1,3,915,58,1,2,1343]", "_type": "bitfield", "elemcount": "2349"}, "Deadline": 6, "Partition": 0}]}`

I would have made this struct match the existing schema:

```go
var ret = struct {
  Count uint64 `json:elemcount`
  RLE []uint64 `json:rle`
  Type string `json:_type`
}{ Type: "bitfield" }
```